Bottom: 04a416cc2d80d7927a2904c5e9dd7b19d8f905d4
Top:    6f185dfedff28cd00412ecf46747662c87a6311d
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-09 17:30:12 +0000

Add the drc flag as a parameter

Ok, this is nasty.. I'm obscuring setting of the global varaible... be careful!


---

diff --git a/src/find.c b/src/find.c
index 1c02bb3..1ef631b 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3327,9 +3327,10 @@ struct drc_info
 static void
 start_do_it_and_dump (int type, void *ptr1, void *ptr2, void *ptr3,
                       int flag, bool AndDraw,
-                      Coord bloat)
+                      Coord bloat, bool is_drc)
 {
   Bloat = bloat;
+  drc = is_drc;
   ListStart (type, ptr1, ptr2, ptr3, flag);
   DoIt (flag, true, AndDraw);
   DumpList ();
@@ -3351,7 +3352,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
 
   if (PCB->Shrink != 0)
     {
-      start_do_it_and_dump (What, ptr1, ptr2, ptr3, DRCFLAG | SELECTEDFLAG, false, -PCB->Shrink);
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, DRCFLAG | SELECTEDFLAG, false, -PCB->Shrink, false);
       /* ok now the shrunk net has the SELECTEDFLAG set */
       ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
       Bloat = 0;
@@ -3362,10 +3363,8 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
           /* make the flag changes undoable */
           ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
           User = true;
-          drc = false;
-          start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true, -PCB->Shrink);
-          drc = true;
-          start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true, 0);
+          start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true, -PCB->Shrink, false);
+          start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true, 0, true);
           User = false;
           drc = false;
           drcerr_count++;
@@ -3395,9 +3394,8 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       DumpList ();
     }
   /* now check the bloated condition */
-  drc = false;
   ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
-  start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, false, 0);
+  start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, false, 0, false);
   flag = FOUNDFLAG;
   ListStart (What, ptr1, ptr2, ptr3, flag);
   Bloat = PCB->Bloat;
@@ -3408,10 +3406,10 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       /* make the flag changes undoable */
       ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
       User = true;
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true, 0, false);
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true, PCB->Bloat, true);
+      User = false;
       drc = false;
-      start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true, 0);
-      drc = true;
-      start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true, PCB->Bloat);
       drcerr_count++;
       LocateError (&x, &y);
       BuildObjectList (&object_count, &object_id_list, &object_type_list);
@@ -3430,15 +3428,13 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       pcb_drc_violation_free (violation);
       free (object_id_list);
       free (object_type_list);
-      User = false;
-      drc = false;
       if (!throw_drc_dialog())
         return (true);
       IncrementUndoSerialNumber ();
       Undo (true);
       /* highlight the rest of the encroaching net so it's not reported again */
       flag = FOUNDFLAG | SELECTEDFLAG;
-      start_do_it_and_dump (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, flag, true, 0);
+      start_do_it_and_dump (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, flag, true, 0, false);
       drc = true;
       Bloat = PCB->Bloat;
       ListStart (What, ptr1, ptr2, ptr3, flag);
